common/analysis/lint_waiver.h

```diff
@@ -33,6 +33,7 @@ namespace verible {
 class LintWaiver {
   // Compact set of line numbers.
   using LineSet = IntervalSet<size_t>;
+  using RegexSet = std::set<std::string>;
```

Comment:
Related to another comment, is it worth storing the compiled regex, perhaps in a `std::map`?

Reply:
I am not sure what would be the benefit of a map here, so I implemented that using a vector, is that okay? Or did you have something specific in mind where map would be better?

Reply:
My question was more about whether it makes sense to pair the string of the regex with its compiled form, like a cache.

Reply:
Yeah, that sounds like a good idea, pushed an implementation that does the caching.